### PR TITLE
Add support for ranger 2.0

### PR DESCRIPTION
**Pull request by PedroRossi**
As mentioned in https://github.com/yaooqinn/spark-ranger/issues/4, this PR proposes adding another version of the `RangerAdminClientImpl` with complies with the `RangerAdminClient` interface which has more methods since Ranger 2.0

Changes made:
- added the travis config to build Spark 2.3 and 2.4 using Ranger 2.0
- added maven specific config to specify which Ranger class to use during test-compile
- the `RangerAdminClientImpl` was refactored into a trait so we could extend the original `RangerAdminClient` interface and re-use the trait

### CODE REVIEW

**Comment by PedroRossi** [comment]
@yaooqinn I noticed you already add the travis config on https://github.com/yaooqinn/spark-ranger/pull/7, should I remove this config from my PR then?

**Comment by PedroRossi** [comment]
@yaooqinn I also noticed that the config `ranger.plugin.spark.policy.source.impl` in the file `ranger-spark-security.xml` is not being used on the tests (I tried using a random name and the tests were still ok) and I also noticed that you have a sample policy, how are the tests working with it? I didn't understand this part well

File: pom.xml [hunk 1]
```diff
@@ -529,6 +532,19 @@
                 <ranger.version>2.0.0</ranger.version>
                 <solr.version>7.7.1</solr.version>
             </properties>
+            <build>
```
Comment (yaooqinn) [inline pom.xml]: can we use `build-helper-maven-plugin` to `add-test-source` instead of excluding  them?

**Comment by yaooqinn** [comment]
> @yaooqinn I also noticed that the config `ranger.plugin.spark.policy.source.impl` in the file `ranger-spark-security.xml` is not being used on the tests (I tried using a random name and the tests were still ok) and I also noticed that you have a sample policy, how are the tests working with it? I didn't understand this part well

Hmm... I don' t remember well. So can we just rm this conf and the class to see if it works well?

**Comment by PedroRossi** [comment]
We can rm the class but not the conf, I already tested it, but not for all of the Ranger versions.

Going to test this later then send a report here

**Comment by yaooqinn** [comment]
I see. If that config is configured wrong or missed, the `org.apache.ranger.admin.client.RangerAdminRESTClient` will be used.

I guess we just need `class RangerAdminClientImpl extends RangerAdminRESTClient`  to fix the issue here.

File: pom.xml (hunk 1 above)
Comment (PedroRossi) [reply pom.xml]: I didn't test it, going to do it and report here

**Comment by PedroRossi** [comment]
> `RangerAdminRESTClient`

This fixed the issue and it is better because we don't need to change the pom in this case

**Comment by PedroRossi** [comment]
> > @yaooqinn I also noticed that the config `ranger.plugin.spark.policy.source.impl` in the file `ranger-spark-security.xml` is not being used on the tests (I tried using a random name and the tests were still ok) and I also noticed that you have a sample policy, how are the tests working with it? I didn't understand this part well
> 
> Hmm... I don' t remember well. So can we just rm this conf and the class to see if it works well?

It works fine without this conf, we can remove this class and this conf since we do not leverage a Ranger for retrieving the policy

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
Let's keep them, this class helps to debug when we add more policies to test resources.

**Comment by yaooqinn** [comment]
LGTM, could you also update README?

**Comment by yaooqinn** [comment]
merged to master, thanks for your help:)

**Pull request closed without merging**